plugins/shortcode-plugin/includes/block-editor/blocks/random-photos/index.js

```diff
@@ -39,21 +40,11 @@ registerBlockType(name, {
 				type: 'block',
 				blocks: ['core/shortcode'],
 				isMatch: ({ text }) => {
-					return text.startsWith('[twitch_example');
+					return text && text.startsWith('[twitch_example');
 				},
 				transform: ({ text }) => {
-					const attributes = text
-						.replace(/\[twitch_example|]|/g, '') //remove the shortcode tags
-						.replace(/"\s/g, '|')
-						.replace(/"/g, '')
-						.split('|'); //split the attributes
-
-					const atts = {};
-					attributes.map((item) => {
-						const split = item.trim().split('=');
-						atts[[split[0]]] = split[1];
-					});
-					return createBlock(name, atts);
+					const shortcode = attrs( text );
+					return createBlock(name, shortcode.named);
```

Comment:
I have not tested that it works. My code included some processing in between, as my attributes were named differently.